### DISCUSSION TITLE
Why do TGI libs have to use hard-coded font sets?

### DISCUSSION THREAD

**Discussion by Fabrizio-Caruso** [cc65/cc65#1586] [Q&A]
Why do TGI libs have to use hard-coded font sets?
Is there a technical reason why it has to be so for TGI and not for CONIO?

It would be nice to have a way to pass a font set at link time for targets like the Atari Lynx or the Apple 2.

Is there a simple way to do this?

**Comment by oliverschmidt** [comment]
I can't/won't comment on anything but the Apple II. On the Apple II there are two reasons:

1. TGI libs are (at least traditionally) dynamically loaded at runtime. So there's no straightforward way to override a symbol at link time. In general this might be workarounded with a so-called 'library reference'. But such a reference then exsists for all driver of the same class (e.g. TGI) on the same target. And for the Apple II we have beside the hires TGI driver the lores TGI driver. That driver doesn't need a font set (but would have it linked in nevertheless when using a library reference to reference the TGI font set).

2. The Apple II hires TGI driver (at least currently) doesn't use the usual Apple II hires bitmap font set. Instead it uses a rather proprietary vector font set. So you wouldn't have fun at all coming up with some alternative font set.

**Comment by Fabrizio-Caruso** [comment]
@oliverschmidt  Thanks fore explaining the reason.

I am currently more interested in the Atari Lynx target. 
So the main problem is the fact that TGI libs are not usually statically linked but this is not a universal rule.
So a solution for the Atari Lynx is possible where the font set is a standard one (btw I drew some of them in https://github.com/cc65/cc65/pull/958).
